test/integration/base_test_setup.go

```diff
@@ -592,15 +592,24 @@ func GetKeyName(t *testing.T) string {
 func ResetKeys(t *testing.T, ctx contextAPI.ChannelProvider, chaincodeID, value string, keys ...string) {
 	chClient, err := channel.New(ctx)
 	require.NoError(t, err, "Failed to create new channel client for resetting keys")
-	for _, key := range keys {
+	for _, k := range keys {
+		key := k
 		// Synchronous transaction
-		_, e := chClient.Execute(
-			channel.Request{
-				ChaincodeID: chaincodeID,
-				Fcn:         "invoke",
-				Args:        ExampleCCTxSetArgs(key, value),
+		_, e := retry.NewInvoker(retry.New(retry.TestRetryOpts)).Invoke(
```

Comment:
I'm not sure I understand what you're trying to fix here? You basically have a retry invoker inside of a retry invoker.

Reply:
@bstasyszyn , I came into diffrent problems during integration tests, and these problems did not appear everytime. These problems as following:
`2020-08-05T14:17:44.8882001Z [35mintegration-tests_1  |[0m --- FAIL: TestLedgerQueries (14.27s)
2020-08-05T14:17:44.8882416Z [35mintegration-tests_1  |[0m     base_test_setup.go:604: 
2020-08-05T14:17:44.8882831Z [35mintegration-tests_1  |[0m         	Error Trace:	base_test_setup.go:604
2020-08-05T14:17:44.8883295Z [35mintegration-tests_1  |[0m         	            				channel_ledger_test.go:98
2020-08-05T14:17:44.8883742Z [35mintegration-tests_1  |[0m         	Error:      	Received unexpected error:
2020-08-05T14:17:44.8884705Z [35mintegration-tests_1  |[0m         	            	Chaincode status Code: (500) UNKNOWN. Description: error in simulation: failed to execute transaction 218fb6679801724fa0599626daf70f568e07b536219e1de6e06f9802748483fe: invalid invocation: chaincode 'example_cc_fabtest_00' has not been initialized for this version, must call as init first
2020-08-05T14:17:44.8885578Z [35mintegration-tests_1  |[0m         	            	Transaction processing for endorser [peer1.org1.example.com:7151]
2020-08-05T14:17:44.8886159Z [35mintegration-tests_1  |[0m         	            	github.com/hyperledger/fabric-sdk-go/pkg/fab/peer.(*peerEndorser).ProcessTransactionProposal
2020-08-05T14:17:44.8886722Z [35mintegration-tests_1  |[0m         	            		/opt/workspace/fabric-sdk-go/pkg/fab/peer/peerendorser.go:108
2020-08-05T14:17:44.8887275Z [35mintegration-tests_1  |[0m         	            	github.com/hyperledger/fabric-sdk-go/pkg/fab/peer.(*Peer).ProcessTransactionProposal
2020-08-05T14:17:44.8887779Z [35mintegration-tests_1  |[0m         	            		/opt/workspace/fabric-sdk-go/pkg/fab/peer/peer.go:216
2020-08-05T14:17:44.8888284Z [35mintegration-tests_1  |[0m         	            	github.com/hyperledger/fabric-sdk-go/pkg/fab/txn.SendProposal.func1
2020-08-05T14:17:44.8888777Z [35mintegration-tests_1  |[0m         	            		/opt/workspace/fabric-sdk-go/pkg/fab/txn/proposal.go:122
2020-08-05T14:17:44.8889199Z [35mintegration-tests_1  |[0m         	            	runtime.goexit
2020-08-05T14:17:44.8889620Z [35mintegration-tests_1  |[0m         	            		/usr/local/go/src/runtime/asm_amd64.s:1373
2020-08-05T14:17:44.8890302Z [35mintegration-tests_1  |[0m         	Test:       	TestLedgerQueries
2020-08-05T14:17:44.8890725Z [35mintegration-tests_1  |[0m         	Messages:   	Failed to reset keys
2020-08-05T14:17:44.8891059Z [35mintegration-tests_1  |[0m FAIL
2020-08-05T14:17:44.8891483Z [35mintegration-tests_1  |[0m FAIL	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/client/ledger	29.189s`
`2020-08-05T14:18:37.5246841Z [35mintegration-tests_1  |[0m --- FAIL: TestEventClient (0.01s)
2020-08-05T14:18:37.5247415Z [35mintegration-tests_1  |[0m     eventclient_test.go:46: Testing Deliver events
2020-08-05T14:18:37.5248042Z [35mintegration-tests_1  |[0m     --- FAIL: TestEventClient/Deliver_Filtered_Block_Events (0.01s)
2020-08-05T14:18:37.5249781Z [35mintegration-tests_1  |[0m         eventclient_test.go:142: CreateAndSendTransactionProposal return error: invoking transaction proposal failed: Transaction processing for endorser [peer1.org1.example.com:7151]: Chaincode status Code: (500) UNKNOWN. Description: error in simulation: failed to execute transaction 41355695064cd196c9e581f2e9ad7b8b4e3ad48829f96b270edf5d96a3304f64: invalid invocation: chaincode 'example_cc_fabtest_09lpjtle5lv' has not been initialized for this version, must call as init first
2020-08-05T14:18:37.5251028Z [35mintegration-tests_1  |[0m FAIL
2020-08-05T14:18:37.5251559Z [35mintegration-tests_1  |[0m FAIL	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/fab	13.926s
2020-08-05T14:18:54.0723116Z [35mintegration-tests_1  |[0m ok  	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/fabsdk/provider	14.683s
2020-08-05T14:19:39.8181667Z [35mintegration-tests_1  |[0m ok  	github.com/hyperledger/fabric-sdk-go/test/integration/pkg/gateway	43.857s
2020-08-05T14:19:39.8182279Z [35mintegration-tests_1  |[0m ?   	github.com/hyperledger/fabric-sdk-go/test/integration/util/runner	[no test files]
2020-08-05T14:19:39.8182672Z [35mintegration-tests_1  |[0m FAIL`
And the reason I guess is the ledger consistency, so I add some retry.

Reply:
OK, so maybe the inner retry can be removed, i.e. channel.WithRetry(retry.DefaultChannelOpts)?

Reply:
I think you are right, and I will add some retryCodes about lifecycle to ResMgmtDefaultRetryableCodes.